tests/unit/test_repos_release.py

```diff
@@ -140,6 +141,25 @@ def test_download(self):
         )
         assert stream.called is False
 
+    def test_download_with_302(self):
+        """Verify the request to download an Asset file."""
+        with mock.patch.object(github3.models.GitHubCore, '_get') as get:
```

Comment:
> I'm not so certain this test is necessary. At least not as a unit test.

@sigmavirus24 , I added this test since our current unit test does not step into this predicate.
https://github.com/sigmavirus24/github3.py/blob/develop/github3/repos/release.py#L230-L243

If I misunderstood, let me know and I'll remove this test case.